net/node/knowaddrlist.go

```diff
@@ -0,0 +1,141 @@
+package node
+
+import (
+	"DNA/common/log"
+	. "DNA/net/protocol"
+	"math/rand"
+	"sync"
+)
+
+const (
+	// needAddressThreshold is the number of addresses under which the
+	// address manager will claim to need more addresses.
+	needAddressThreshold = 1000
+)
+
+type KnownAddress struct {
+	srcAddr NodeAddr
+}
+
+type KnownAddressList struct {
+	sync.RWMutex
+	List      map[uint64]*KnownAddress
+	addrCount uint64
+}
+
+func (ka *KnownAddress) SaveAddr(na NodeAddr) {
+	ka.srcAddr.Time = na.Time
+	ka.srcAddr.Services = na.Services
+	ka.srcAddr.IpAddr = na.IpAddr
+	ka.srcAddr.Port = na.Port
+	ka.srcAddr.ID = na.ID
+}
+
+func (ka *KnownAddress) NetAddress() NodeAddr {
```

Comment:
use `NodeAddr` directly or `type KnownAddress NodeAddr`  to avoid redundant code ?

Reply:
👌